### ISSUE TITLE
Allow asserting, regardless of whether i.e. a Throwable is present

### ISSUE THREAD

**Issue by jamietanna** [valfirst/slf4j-test#188]
Current implementations expect full equality checks, including `arguments`, `throwable, `mdc`, etc to match.

One alternative is to create i.e.:

```java
class NonCheckingThrowable extends Throwable {
  @Override
  public boolean equals(Object other) {
    return true;
  }
}
```

But this is not ideal - we should have some way of configuring this

**Comment by josephw** [comment]
Are you using `assertEquals`, and requiring a `LoggingEvent` that's going to partially match? Would you consider [Hamcrest-style matchers](http://hamcrest.org/JavaHamcrest/javadoc/2.2/org/hamcrest/Matchers.html) instead:

    assertThat(logEvent, messageMatches(...));

or [AssertJ custom assertions](https://assertj.github.io/doc/#assertj-core-custom-assertions):

    assertThat(logEvent).hasMessageMatching(...);

? Those seem like something that could live in this, or a related, library.

**Comment by valfirst** [comment]
Fixed: #328
